### PR TITLE
Add support for PrincipalIdentity +1

### PR DESCRIPTION
**Pull request by TaoZou1**
Add support for PrincipalIdentity.  
Log unsupported type.  
Return error if CasttoPointer result is nil

### CODE REVIEW

File: pkg/nsx/util/utils.go [hunk 1]
```diff
@@ -547,6 +548,8 @@ func FindTag(tags []model.Tag, tagScope string) string {
 
 func CasttoPointer(obj interface{}) interface{} {
```
Comment (gran-vmv) [inline pkg/nsx/util/utils.go:549]: Maybe you can try:
```
func PtrTo[T any](v T) *T {
  return &v
}
```

File: pkg/nsx/util/utils.go [hunk 2]
```diff
@@ -578,6 +581,8 @@ func CasttoPointer(obj interface{}) interface{} {
 	case model.IpAddressBlock:
 		return &v
 	default:
+		objType := reflect.TypeOf(obj)
+		log.Info("Unsupported type", "objType", objType.String())
```
[the remark this answers is not in the document]
Comment (TaoZou1) [reply pkg/nsx/util/utils.go]: removed.
the objType can't be nil. since obj won't be nil.

File: pkg/nsx/util/utils.go (hunk 1 above)
Comment (TaoZou1) [reply pkg/nsx/util/utils.go]: amazing!
Update it.

**Comment by gran-vmv** [comment]
LGTM

**Review by zhengxiexie — APPROVED** [review]
LGTM

**Review by gran-vmv — APPROVED** [review]

**Pull request merged**